httpx/_client.py

```diff
@@ -1105,7 +1107,7 @@ def close(self) -> None:
                 if proxy is not None:
                     proxy.close()
 
-    def __enter__(self) -> "Client":
+    def __enter__(self) -> Self:
```

Comment:
I *think* something like this would do:

```
T = TypeVar("T")

def __enter__(self: T) -> T: ...
```

Reply:
yup, T and Self are functionally equivalent but just have different names